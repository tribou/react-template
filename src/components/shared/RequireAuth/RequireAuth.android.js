// @flow
import RequireAuth from './RequireAuth.ios'

export default RequireAuth
