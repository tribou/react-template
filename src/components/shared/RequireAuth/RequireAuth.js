// @flow
import React, { PureComponent } from 'react'
import { Redirect } from 'react-router'
import type { Location } from 'react-router'
import type { Node } from 'react'

import type { ReduxProps } from './'


type Props = ReduxProps & {
  children?: Node,
  location: Location,
  to?: string,
}

class RequireAuth extends PureComponent<Props> {

  // TODO: watch for resolution of
  // https://github.com/yannickcr/eslint-plugin-react/issues/1376
  props: Props

  render () {

    const { children, location, to, token } = this.props
    const { hash, pathname, search } = location
    const redirect = encodeURIComponent(`${pathname}${search}${hash}`)

    const t = to || `/home?m=login&redirect=${redirect}`

    if (!token && !children) return <Redirect to={t} />
    if (!token && children) return null

    return children || null

  }

}

export default RequireAuth
