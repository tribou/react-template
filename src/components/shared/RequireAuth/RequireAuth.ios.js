// @flow
import RequireAuth from './RequireAuth'

export default RequireAuth
