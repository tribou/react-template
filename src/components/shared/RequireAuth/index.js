// @flow
import { connect } from "react-redux";
import { withRouter } from "react-router";
import type { RootReducerState } from "src/redux/modules";
import RequireAuth from "./RequireAuth";

type StateProps = {
  token: ?string
};

const mapStateToProps = ({
  auth: { token }
}: RootReducerState): StateProps => ({ token });

export default withRouter(connect(mapStateToProps)(RequireAuth));
export type ContainerProps = StateProps;
