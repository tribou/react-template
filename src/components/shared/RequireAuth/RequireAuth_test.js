// @flow

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import RequireAuth from './RequireAuth'


const mockProps = {
  location: {
    hash: '',
    search: '',
    pathname: '',
  },
  token: undefined,
  to: undefined,
  children: undefined,
}


it('renders <Redirect> if no token', () => {

  const wrapper = shallow(
    <RequireAuth
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})


it('passes {to} prop to <Redirect> if no token', () => {

  const to = '/go/here/if/no/auth'
  const wrapper = shallow(
    <RequireAuth
      {...mockProps}
      to={to}
    />
  )

  expect(wrapper.find('Redirect').prop('to')).toEqual(to)

})


it('renders children if token is present', () => {

  const wrapper = shallow(
    <RequireAuth
      {...mockProps}
      token="test"
    >
      <button>Logout</button>
    </RequireAuth>
  )

  expect(wrapper.find('button').text()).toEqual('Logout')

})


it('renders nothing if children and token is not present', () => {

  const wrapper = shallow(
    <RequireAuth
      {...mockProps}
    >
      <button>Logout</button>
    </RequireAuth>
  )

  expect(wrapper.getElement()).toEqual(null)

})


it('renders nothing if no children and token is present', () => {

  const wrapper = shallow(
    <RequireAuth
      {...mockProps}
      token="test"
    />
  )

  expect(wrapper.getElement()).toEqual(null)

})
