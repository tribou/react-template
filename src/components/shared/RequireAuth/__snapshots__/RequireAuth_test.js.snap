// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Redirect> if no token 1`] = `
<Redirect
  push={false}
  to="/home?m=login&redirect="
/>
`;
